specs/MultiProfile.md

```diff
@@ -169,20 +169,37 @@ void ScenarioCookieManagement::DeleteAllCookies()
 ### Delete profile
 
 ```cpp
-HRESULT AppWindow::DeleteProfile(ICoreWebView2Controller* controller)
+HRESULT AppWindow::DeleteProfile(ICoreWebView2* webView2)
 {
-    wil::com_ptr<ICoreWebView2> coreWebView2;
-    CHECK_FAILURE(controller->get_CoreWebView2(&coreWebView2));
-    auto webview7 = coreWebView2.try_query<ICoreWebView2_7>();
-    if (webview7)
+    DCHECK(webView2);
+    wil::com_ptr<ICoreWebView2Profile> profile;
+    CHECK_FAILURE(webView2->get_Profile(&profile));
+    CHECK_FAILURE(profile2->Delete());
+}
+
+void AppWindow::RegisterProfileDeletedEventHandlers(ICoreWebView2* webView2)
+{
+    wil::com_ptr<ICoreWebView2Profile> profile;
+    CHECK_FAILURE(webView2->get_Profile(&profile));
+    if (profile)
     {
-        wil::com_ptr<ICoreWebView2Profile> profile;
-        CHECK_FAILURE(webview7->get_Profile(&profile));
-        auto profile2 = profile.try_query<ICoreWebView2StagingProfile4>();
-        if (profile2)
-        {
-            CHECK_FAILURE(profile2->Delete());
-        }
+        CHECK_FAILURE(profile->add_Deleted(
+            Microsoft::WRL::Callback<ICoreWebView2StagingProfileDeletedEventHandler>(
+                [this](ICoreWebView2Profile* sender, IUnknown* args)
+                {
+                    RunAsync(
+                        [this]()
+                        {
+                            std::wstring message = L"The webview2 has been closed and "
```

Comment:
This event is on the Profile, and the sample has been updated so that it doesn't associate the profile with a webview2 at all. The message shown to the user is misleading. There may not be a webview2 at all.

Maybe

> The profile has been marked for deletion. Any associated webview2 objects will be closed.

Reply:
ok, will update to this.